### DISCUSSION TITLE
Duda sobre complejidad al imprimir en parte 2

### DISCUSSION THREAD

**Discussion by gaspar-mr05** [IIC2133-PUC/2024-1#102] [Tarea 1] [answered]
Hola! Tengo dudas sobre la impresión de la cantidad de encontrados en la parte 2 en general. Que pasa si yo itero sobre las cabezas que ya encontré solo para contarlas y hacer print de la cantidad total que encontré, y luego ya itero para imprimir el nombre y el id. El iterar para contar la cantidad total de estas cuenta para el calculo de la complejidad? Tengo entendido que como "busqueda" se considerara el momento en el que se itera pero para imprimir nombre y id, en mi caso, tengo que cada valor esta asociado a una cola ordenada por prioridad, por lo mismo debo "iterar" sobre la cola de tamaño k, con k cabezas encontradas.

**Comment by sebacarrascop** [comment, marked as answer]
Hola!, claro, ya que al hacer la búsqueda la haces sobre N elementos, donde consigues k resultados, luego iterar sobre estos k resultados suma una complejidad O(k), que sumado a la complejidad de la búsqueda de N elementos resulta despreciable, por lo que la complejidad se mantendría. Puedes ver sobre esto también en #97
